core/server/utils/read-csv.js

```diff
@@ -0,0 +1,58 @@
+var readline = require('readline'),
+    Promise = require('bluebird'),
+    lodash = require('lodash'),
+    errors = require('../errors'),
+    fs = require('fs');
+
+function readCSV(options) {
+    var path = options.path,
+        columnsToExtract = options.columnsToExtract || [],
+        firstLine = true,
+        mapping = {},
+        toReturn = [],
+        rl;
+
+    return new Promise(function (resolve, reject) {
+        rl = readline.createInterface({
+            input: fs.createReadStream(path),
+            terminal: false
+        });
+
+        rl.on('line', function (line) {
+            var values = line.split(','),
+                entry = {};
+
+            // CASE: column headers
+            if (firstLine) {
+                try {
+                    lodash.each(columnsToExtract, function (columnToExtract) {
+                        mapping[columnToExtract] = values.indexOf(columnToExtract);
+
+                        // CASE: column does not exist
+                        if (mapping[columnToExtract] === -1) {
+                            throw new errors.ValidationError(
+                                'Column header missing: "{{column}}".'.replace('{{column}}', columnToExtract)
+                            );
+                        }
```

Comment:
Cool, the different column names are working now, but it doesn't work if I only provide a single column with no header.